rolepermissions/roles.py

```diff
@@ -49,24 +49,30 @@ def get_name(cls):
     @classmethod
     def assign_role_to_user(cls, user):
         """
-        Deletes all of user's previous roles, and removes all permissions
-        mentioned in their available_permissions property.
+        Assign this role to a user.
 
         :returns: :py:class:`django.contrib.auth.models.Group` The group for the
             new role.
         """
-        from rolepermissions.shortcuts import remove_role
-
-        remove_role(user)
-
-        group, created = Group.objects.get_or_create(name=cls.get_name())
+        group, _created = Group.objects.get_or_create(name=cls.get_name())
         user.groups.add(group)
         permissions_to_add = cls.get_default_true_permissions()
         user.user_permissions.add(*permissions_to_add)
 
         return group
 
     @classmethod
+    def remove_role_from_user(cls, user):
```

Comment:
I think we should be more considerative of the previous state of the permissions. Here are some user stories to consider:

Doctor:
- enter_surgery_room: False

Surgeon:
- enter_surgery_room: True

Anestezist:
- enter_surgery_room: True

user is `Doctor` and `Surgeon` and can enter_surgery_room
Remove `Surgeon` role
user canNOT enter_surgery_room

user is `Doctor` and `Surgeon` and can enter_surgery_room
Remove `Doctor` role
user can enter_surgery_room

user is `Doctor` and `Surgeon` and canNOT enter_surgery_room
Remove `Surgeon` role
user canNOT enter_surgery_room

user is `Doctor` and `Surgeon` and canNOT enter_surgery_room
Remove `Doctor` role
user canNOT enter_surgery_room

user is `Doctor` and `Surgeon` and `Anestezist` and can enter_surgery_room
Remove `Surgeon` role
user can enter_surgery_room

user is `Doctor` and `Surgeon` and `Anestezist` and can enter_surgery_room
Remove `Doctor` role
user can enter_surgery_room

user is `Doctor` and `Surgeon` and `Anestezist` and canNOT enter_surgery_room
Remove `Surgeon` role
user canNOT enter_surgery_room

user is `Doctor` and `Surgeon` and `Anestezist` and canNOT enter_surgery_room
Remove `Doctor` role
user canNOT enter_surgery_room